src/plugins/chart_expressions/expression_xy/common/helpers/table.ts

```diff
@@ -15,16 +15,14 @@ export function normalizeTable(data: Datatable, xAccessor?: string | ExpressionV
   const xColumn = xAccessor && getColumnByAccessor(xAccessor, data.columns);
   if (xColumn && xColumn?.meta.type === 'date') {
     const xColumnId = xColumn.id;
-    const rows = data.rows.reduce<Datatable['rows']>((normalizedRows, row) => {
-      return [
-        ...normalizedRows,
-        {
-          ...row,
-          [xColumnId]:
-            typeof row[xColumnId] === 'string' ? moment(row[xColumnId]).valueOf() : row[xColumnId],
-        },
-      ];
-    }, []);
+    if (!data.rows.some((row) => typeof row[xColumnId] === 'string')) return data;
+    const rows = data.rows.map((row) => {
+      return {
+        ...row,
+        [xColumnId]:
+          typeof row[xColumnId] === 'string' ? moment(row[xColumnId]).valueOf() : row[xColumnId],
+      };
```

Comment:
At this point I'd argue to reduce even more the amount of work:

```suggestion
    const rows = data.rows.map((row) => {
      return typeof row[xColumnId] !== 'string' ? row : {
          ...row,
          [xColumnId]: moment(row[xColumnId]).valueOf(),
        };
      }
```